### PR TITLE
Added a script to build a docker image for candid.

### PR DESCRIPTION
**Pull request by alesstimec**
## Description

Adds a script to build a candid docker image and push it to the specified registry.

## Engineering checklist
*Check only items that apply*

- [ ] Documentation updated
- [ ] Covered by unit tests
- [ ] Covered by integration tests
- [ ] Independent change*

### CODE REVIEW

File: scripts/image-build.sh [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+fi
+
+VERSION=`git describe --dirty --always`
+GIT_COMMIT=`git rev-parse --verify HEAD`
+
+docker build \
+       --build-arg http_proxy \
+       --build-arg https_proxy \
+       --build-arg no_proxy \
+       --build-arg NO_PROXY \
```
Comment (andrei-polukhin) [inline scripts/image-build.sh:27]: Do we need both of them?

**Review by andrei-polukhin — APPROVED** [review]
LGTM, but someone else must approve

File: scripts/image-build.sh [hunk 2]
```diff
@@ -0,0 +1,32 @@
+
+#!/bin/bash
+
+# This script builds the candid docker image and pushes it to the
```
Comment (ale8k) [inline scripts/image-build.sh:4]: Java world 1, rest of us 0: https://github.com/GoogleContainerTools/jib :D jokes aside LGTM but would suggest buildkit for secrets & keys

File: scripts/image-build.sh [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+VERSION=`git describe --dirty --always`
+GIT_COMMIT=`git rev-parse --verify HEAD`
+
+docker build \
+       --build-arg http_proxy \
+       --build-arg https_proxy \
+       --build-arg no_proxy \
+       --build-arg NO_PROXY \
+       -t ${DOCKER_REGISTRY}/candid:${VERSION} \
```
Comment (ale8k) [inline scripts/image-build.sh:28]: Would prefer jenkins semver manager plugin and/or have a consolidated build versioning strategy (to include canary, snapshot, etc.) But LGTM if this is our approach.

**Review by ale8k — APPROVED** [review]

File: scripts/image-build.sh [hunk 4]
```diff
@@ -0,0 +1,32 @@
+
+#!/bin/bash
+
+# This script builds the candid docker image and pushes it to the
+# configured docker registry,
+# Required environment variables:
+# - DOCKER_REGISTRY
```
Comment (ale8k) [inline scripts/image-build.sh:7]: Should we push these as arguments to the script? Prevents us requiring env vars then

File: scripts/image-build.sh [hunk 5]
```diff
@@ -0,0 +1,32 @@
+
```
Comment (ale8k) [inline scripts/image-build.sh:1]: Extra line here :P & maybe make target be cool

File: scripts/image-build.sh [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+docker build \
+       --build-arg http_proxy \
+       --build-arg https_proxy \
+       --build-arg no_proxy \
+       --build-arg NO_PROXY \
+       -t ${DOCKER_REGISTRY}/candid:${VERSION} \
+       -f ./Dockerfile .
+docker tag ${DOCKER_REGISTRY}/candid:${VERSION} ${DOCKER_REGISTRY}/candid:latest
+docker push ${DOCKER_REGISTRY}/candid:${VERSION}
+docker push ${DOCKER_REGISTRY}/candid:latest
```
Comment (ale8k) [inline scripts/image-build.sh:32]: New line : D

**Review by brotich — APPROVED** [review]
LGTM, the other comments here raise a good point for discussion

**Pull request merged**